s3transfer/manager.py

```diff
@@ -273,6 +293,46 @@ def _validate_all_known_args(self, actual, allowed):
                     "must be one of: %s" % (
                         kwarg, ', '.join(allowed)))
 
+    def _submit_transfer(self, call_args, submission_task_cls,
+                         extra_main_kwargs=None):
+        if not extra_main_kwargs:
+            extra_main_kwargs = {}
+
+        # Create a TransferFuture to return back to the user
+        transfer_future, components = get_transfer_future_with_components(
```

Comment:
Not really. I can move it to this class. There is not really a reason it needs to be external other than it is a little easier to test, but should not be an issue.